// @chevtek/poker-engine/build/Card.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardRank = exports.CardSuit = exports.CardColor = exports.Card = void 0;
class Card {
    constructor(rank, suit) {
        this._rank = rank;
        this._suit = suit;
    }
    get rank() {
        return this._rank;
    }
    get suit() {
        return this._suit;
    }
    get color() {
        switch (this._suit) {
            case CardSuit.CLUB:
            case CardSuit.SPADE:
                return CardColor.BLACK;
            case CardSuit.DIAMOND:
            case CardSuit.HEART:
                return CardColor.RED;
        }
    }
    get suitChar() {
        switch (this._suit) {
            case CardSuit.CLUB:
                return "♣";
            case CardSuit.DIAMOND:
                return "♦";
            case CardSuit.HEART:
                return "♥";
            case CardSuit.SPADE:
                return "♠";
        }
    }
}
exports.Card = Card;
var CardColor;
(function (CardColor) {
    CardColor["RED"] = "#ff0000";
    CardColor["BLACK"] = "#000000";
})(CardColor = exports.CardColor || (exports.CardColor = {}));
var CardSuit;
(function (CardSuit) {
    CardSuit["CLUB"] = "c";
    CardSuit["DIAMOND"] = "d";
    CardSuit["HEART"] = "h";
    CardSuit["SPADE"] = "s";
})(CardSuit = exports.CardSuit || (exports.CardSuit = {}));
var CardRank;
(function (CardRank) {
    CardRank["ACE"] = "A";
    CardRank["KING"] = "K";
    CardRank["QUEEN"] = "Q";
    CardRank["JACK"] = "J";
    CardRank["TEN"] = "T";
    CardRank["NINE"] = "9";
    CardRank["EIGHT"] = "8";
    CardRank["SEVEN"] = "7";
    CardRank["SIX"] = "6";
    CardRank["FIVE"] = "5";
    CardRank["FOUR"] = "4";
    CardRank["THREE"] = "3";
    CardRank["TWO"] = "2";
})(CardRank = exports.CardRank || (exports.CardRank = {}));
